linera-storage/src/migration.rs

```diff
@@ -155,40 +158,49 @@ where
     }
 
     pub async fn migrate_if_needed(&self) -> Result<(), ViewError> {
-        let (is_initialized, schema) = self.get_storage_state().await?;
-        if is_initialized && schema == SchemaVersion::Version0 {
-            self.migrate_v0_to_v1().await?;
+        loop {
+            if matches!(
+                self.get_storage_state().await?,
+                None | Some(SchemaVersion::Version1)
+            ) {
+                // Nothing to do.
+                return Ok(());
+            }
+            let result = self.migrate_v0_to_v1().await;
+            if let Err(ViewError::MissingEntries) = result {
+                tracing::warn!(
+                    "It looks like a migration is already in progress on this database. \
+                     I will wait for {:?} minutes and retry.",
+                    MIGRATION_WAIT_BEFORE_RETRY_MIN
+                );
+                // Duration::from_mins is not yet stable for tokio 1.36.
+                tokio::time::sleep(Duration::from_secs(MIGRATION_WAIT_BEFORE_RETRY_MIN * 60)).await;
```

Comment:
Unfortunately, the `from_mins` seems to be part of unstable (how hard could it be?).

Reply:
yes, that's what I wrote in the comment